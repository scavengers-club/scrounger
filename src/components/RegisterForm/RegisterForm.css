.register-form {
  display: flex;
  flex-direction: column;
  gap: 2vh;
  width: 80vw;
  margin: 10%;
  align-items: center;
}

.register-input {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.register-input input {
  background-color: rgba(255, 255, 255, 0.527);
  color: black;
  border-radius: 7px;
  padding: 3px;
  text-align: center;
}

.register-button {
  width: 20vw;
  background-color: var(--purple);
  border-radius: 7px;
  padding: 3px;
}

fieldset img {
  width: 15vw;
  margin: 1vw;
}

fieldset input[type='radio'] {
  display: none;
}

fieldset input[type='radio']:checked + img {
  filter: drop-shadow(0px 10px 10px var(--sand));
}

.register-form input::placeholder {
  color: var(--purple);
}

.redirect {
  text-align: center;
}
