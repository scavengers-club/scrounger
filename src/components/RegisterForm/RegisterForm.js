import { Link } from 'react-router-dom';

import './RegisterForm.css';

export default function RegisterForm({
  username,
  setUsername,
  email,
  setEmail,
  password,
  setPassword,
  checked,
  setChecked,
  avatars,
  setSelectedAvatar,
  handleSubmit,
}) {
  return (
    <>
      <form className="register-form">
        <label className="register-input">
          username:
          <input
            type="text"
            value={username}
            onChange={(e) => setUsername(e.target.value)}
            placeholder="make it unique"
          />
        </label>
        <label className="register-input">
          email:
          <input
            type="email"
            value={email}
            onChange={(e) => setEmail(e.target.value)}
            placeholder="make it real"
          />
        </label>
        <label className="register-input">
          password:
          <input
            type="password"
            value={password}
            onChange={(e) => setPassword(e.target.value)}
            placeholder="make it strong"
          />
        </label>

        <fieldset>
          <legend>Avatar:</legend>
          <label>
            <input
              type="radio"
              name="avatar"
              value={avatars[0]}
              onChange={(e) => setSelectedAvatar(e.target.value)}
            />
            <img src={`${process.env.PUBLIC_URL}/avatars/${avatars[0]}`} />
          </label>
          <label>
            <input
              type="radio"
              name="avatar"
              value={avatars[1]}
              onChange={(e) => setSelectedAvatar(e.target.value)}
            />
            <img src={`${process.env.PUBLIC_URL}/avatars/${avatars[1]}`} />
          </label>
          <label>
            <input
              type="radio"
              name="avatar"
              value={avatars[2]}
              onChange={(e) => setSelectedAvatar(e.target.value)}
            />
            <img src={`${process.env.PUBLIC_URL}/avatars/${avatars[2]}`} />
          </label>
          <label>
            <input
              type="radio"
              name="avatar"
              value={avatars[3]}
              onChange={(e) => setSelectedAvatar(e.target.value)}
            />
            <img src={`${process.env.PUBLIC_URL}/avatars/${avatars[3]}`} />
          </label>
        </fieldset>

        <label>
          <input type="checkbox" value={checked} onChange={() => setChecked(!checked)} />
          <span>I am over the age of 13</span>
        </label>

        <button className="register-button" onClick={handleSubmit}>
          Register
        </button>
      </form>
      <div className="redirect">
        <p>Already have an account?</p>
        <Link to="/login">Log In</Link>
      </div>
    </>
  );
}
