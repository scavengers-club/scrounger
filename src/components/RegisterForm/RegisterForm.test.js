import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import RegisterForm from './RegisterForm';

test('<RegisterForm/> should display an empty form for new user', () => {
  const avatars = [`westerner-hat.png`, `safety-hat.png`, `chullo-hat.png`, `house-hat.png`];
  const { container } = render(
    <MemoryRouter initialEntries={['/register']}>
      <Route path="/register">
        <RegisterForm avatars={avatars} />
      </Route>
    </MemoryRouter>
  );

  screen.getAllByDisplayValue('');
  expect(container).toMatchSnapshot();
});
