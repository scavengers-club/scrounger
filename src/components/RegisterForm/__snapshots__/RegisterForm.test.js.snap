// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegisterForm/> should display an empty form for new user 1`] = `
<div>
  <form
    class="register-form"
  >
    <label
      class="register-input"
    >
      username:
      <input
        placeholder="make it unique"
        type="text"
        value=""
      />
    </label>
    <label
      class="register-input"
    >
      email:
      <input
        placeholder="make it real"
        type="email"
        value=""
      />
    </label>
    <label
      class="register-input"
    >
      password:
      <input
        placeholder="make it strong"
        type="password"
        value=""
      />
    </label>
    <fieldset>
      <legend>
        Avatar:
      </legend>
      <label>
        <input
          name="avatar"
          type="radio"
          value="westerner-hat.png"
        />
        <img
          src="/avatars/westerner-hat.png"
        />
      </label>
      <label>
        <input
          name="avatar"
          type="radio"
          value="safety-hat.png"
        />
        <img
          src="/avatars/safety-hat.png"
        />
      </label>
      <label>
        <input
          name="avatar"
          type="radio"
          value="chullo-hat.png"
        />
        <img
          src="/avatars/chullo-hat.png"
        />
      </label>
      <label>
        <input
          name="avatar"
          type="radio"
          value="house-hat.png"
        />
        <img
          src="/avatars/house-hat.png"
        />
      </label>
    </fieldset>
    <label>
      <input
        type="checkbox"
        value=""
      />
      <span>
        I am over the age of 13
      </span>
    </label>
    <button
      class="register-button"
    >
      Register
    </button>
  </form>
  <div
    class="redirect"
  >
    <p>
      Already have an account?
    </p>
    <a
      href="/login"
    >
      Log In
    </a>
  </div>
</div>
`;
